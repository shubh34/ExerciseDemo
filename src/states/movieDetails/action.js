import { RSAA } from 'redux-api-middleware';
import endPoints from '../../configs/endPoints';


export const GET_MOVIE_DETAILS_REQUEST = 'GET_MOVIE_DETAILS_REQUEST';
export const GET_MOVIE_DETAILS_SUCCESS = 'GET_MOVIE_DETAILS_SUCCESS';
export const GET_MOVIE_DETAILS_FAILURE = 'GET_MOVIE_DETAILS_FAILURE';

export const RESET_MOVIE_DETAILS = 'RESET_MOVIE_DETAILS';

export const getMovieDetails = (movieName) => {
	const endPoint = endPoints.getMovieDetails;
	const url = endPoint.resolve(movieName);
	return {
		[RSAA]: {
			endpoint: url,
			method: endPoint.method,
			types: [GET_MOVIE_DETAILS_REQUEST, GET_MOVIE_DETAILS_SUCCESS, GET_MOVIE_DETAILS_FAILURE],
		},
	};
};

export const resetMovieDetails = () => ({
	type: RESET_MOVIE_DETAILS,
});
