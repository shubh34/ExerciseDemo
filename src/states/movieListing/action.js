import { RSAA } from 'redux-api-middleware';
import endPoints from '../../configs/endPoints';
import { cacheApiResponse } from '../../utils/cacheApiUtils';

export const GET_MOVIE_LIST_BY_CATEGORY_REQUEST = 'GET_MOVIE_LIST_BY_CATEGORY_REQUEST';
export const GET_MOVIE_LIST_BY_CATEGORY_SUCCESS = 'GET_MOVIE_LIST_BY_CATEGORY_SUCCESS';
export const GET_MOVIE_LIST_BY_CATEGORY_FAILURE = 'GET_MOVIE_LIST_BY_CATEGORY_FAILURE';

export const createRSAAMovieListReuest = (categoryName) => {
	const endPoint = endPoints.fetchMovieListByCategory;
	const url = endPoint.resolve(categoryName);
	const success = { type: GET_MOVIE_LIST_BY_CATEGORY_SUCCESS, meta: { id: categoryName, cacheKey: GET_MOVIE_LIST_BY_CATEGORY_REQUEST } };
	const failure = { type: GET_MOVIE_LIST_BY_CATEGORY_FAILURE, meta: { id: categoryName } };
	return {
		[RSAA]: {
			endpoint: url,
			method: endPoint.method,
			types: [GET_MOVIE_LIST_BY_CATEGORY_REQUEST, success, failure],
		},
	};
};

export const fetchMovieListByCategory = categoryName => dispatch => dispatch(createRSAAMovieListReuest(categoryName)).then((response) => {
	cacheApiResponse(response);
});
