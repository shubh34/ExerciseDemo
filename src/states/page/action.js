
export const GO_TO_PAGE = 'GO_TO_PAGE';


export const goToPage = name => ({
	type: GO_TO_PAGE,
	name,
});
