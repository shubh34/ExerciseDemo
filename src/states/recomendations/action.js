import { RSAA } from 'redux-api-middleware';
import endPoints from '../../configs/endPoints';

export const GET_MOVIE_RECOMENDATION_REQUEST = 'GET_MOVIE_RECOMENDATION_REQUEST';
export const GET_MOVIE_RECOMENDATION_SUCCESS = 'GET_MOVIE_RECOMENDATION_SUCCESS';
export const GET_MOVIE_RECOMENDATION_FAILURE = 'GET_MOVIE_RECOMENDATION_FAILURE';

export const fetchRecomendationForMovie = id => {
	const endPoint = endPoints.getRecomendationForMovie;
	const url = endPoint.resolve(id);
	return {
		[RSAA]: {
			endpoint: url,
			method: endPoint.method,
			headers: {
				Accept: 'application/json',
				'content-type': 'application/json'
			},
			types: [GET_MOVIE_RECOMENDATION_REQUEST, GET_MOVIE_RECOMENDATION_SUCCESS, GET_MOVIE_RECOMENDATION_FAILURE]
		}
	};
};
