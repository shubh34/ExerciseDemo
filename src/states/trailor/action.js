import { RSAA } from 'redux-api-middleware';
import endPoints from '../../configs/endPoints';

export const GET_MOVIE_TRAILOR_REQUEST = 'GET_MOVIE_TRAILOR_REQUEST';
export const GET_MOVIE_TRAILOR_SUCCESS = 'GET_MOVIE_TRAILOR_SUCCESS';
export const GET_MOVIE_TRAILOR_FAILURE = 'GET_MOVIE_TRAILOR_FAILURE';


export const getMovieTrailor = () => {
	const endPoint = endPoints.getMovieTrailer;
	const url = endPoint.resolve();
	return {
		[RSAA]: {
			endpoint: url,
			method: endPoint.method,
			headers: {
				Accept: 'application/json',
				'content-type': 'application/json',

			},
			body: JSON.stringify({
				audio_language: 'SPA',
				audio_quality: '2.0',
				content_id: 'matrix',
				content_type: 'movies',
				device_serial: 'device_serial_1',
				device_stream_video_quality: 'FHD',
				player: 'web:PD-NONE',
				subtitle_language: 'MIS',
				video_type: 'trailer',
			}),
			types: [GET_MOVIE_TRAILOR_REQUEST, GET_MOVIE_TRAILOR_SUCCESS, GET_MOVIE_TRAILOR_FAILURE],
		},
	};
};
