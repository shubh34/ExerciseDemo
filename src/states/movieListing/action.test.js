import { getMovieListByCategory } from './action';


describe('movie list action', () => {
	it('should create action for getting movie list', () => {
		expect(getMovieListByCategory('test')).toMatchSnapshot();
	});
});
