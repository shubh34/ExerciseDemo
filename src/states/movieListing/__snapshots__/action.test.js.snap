// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`movie list action should create action for getting movie list 1`] = `[Function]`;
